includes/sanitizers/class-amp-object-sanitizer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Class AMP_Object_Sanitizer
+ *
+ * @package AMP
+ */
+
+use AmpProject\Attribute;
+use AmpProject\Layout;
+
+/**
+ * Class AMP_Object_Sanitizer
+ *
+ * Sanitizes `<object>` embeds.
+ *
+ * @since 2.1
+ * @internal
+ */
+class AMP_Object_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Default PDF embed height.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_PDF_EMBED_HEIGHT = 600;
+
+	/**
+	 * Sanitize `object` elements from the HTML contained in this instance's Dom\Document.
+	 */
+	public function sanitize() {
+		$elements = $this->dom->getElementsByTagName( 'object' );
+
+		if ( 0 === $elements->length ) {
+			return;
+		}
+
+		/** @var DOMElement $element */
+		foreach ( $elements as $element ) {
+			if ( $element->getAttribute( Attribute::TYPE ) === 'application/pdf' ) {
+				$this->sanitize_pdf( $element );
+			}
+		}
+	}
+
+	/**
+	 * Sanitize PDF embeds.
+	 *
+	 * @see \AMP_Core_Block_Handler::ampify_file_block()
+	 *
+	 * @param DOMElement $element Object element.
+	 */
+	public function sanitize_pdf( DOMElement $element ) {
```

Comment:
The code can now make use of the extended `AmpProject\Dom\Element`.
```suggestion
	 * @param Element $element Object element.
	 */
	public function sanitize_pdf( Element $element ) {
```